File: sources/java/03_func_sum/Main.java
// task 03 func_sum — expected output: 100000000
// build: javac Main.java    run: java Main
// build (graalvm native-image): native-image -O2 Main    run: ./Main
// The JVM has no portable no-inline attribute, so addOne is a plain static method;
// HotSpot may still inline it, which would make this task measure only the loop.

public class Main {
    private static long addOne(long n) {
        return n + 1;
    }

    public static void main(String[] args) {
        long value = 0;
        for (long i = 0; i < 100000000L; i++) {
            value = addOne(value);
        }
        System.out.println(value);
    }
}

File: sources/php/03_func_sum/main.php
<?php
// task 03 func_sum — expected output: 100000000
// build: none (interpreted)    run: php main.php (zend) | php -d opcache.enable_cli=1 -d opcache.jit_buffer_size=64M main.php (zend + jit)
// opcache and the JIT may inline add_one; the loop still makes 100000000 calls either way.

function add_one(int $n): int
{
    return $n + 1;
}

$value = 0;

for ($i = 0; $i < 100000000; $i++) {
    $value = add_one($value);
}

echo $value, "\n";
